src/mpi/comm/comm_impl.c

```diff
@@ -801,6 +801,10 @@ int MPIR_Intercomm_create_from_groups_impl(MPIR_Group * local_group_ptr, int loc
 
 int MPIR_Comm_free_impl(MPIR_Comm * comm_ptr)
 {
+    if (comm_ptr == MPIR_Process.comm_parent) {
+        /* We only release comm_parent in MPI_Finalize */
+        return MPI_SUCCESS;
+    }
```

Comment:
Is this the implementation reached by `MPI_Comm_disconnect`? Will it still satisfy this requirement from MPI-4.0 page 527?
> After the parent communicator is freed or disconnected, `MPI_COMM_GET_PARENT` returns `MPI_COMM_NULL`.

Reply:
Not really. I'll make another fix